### PR TITLE
Factored out DataDirectory string replacements.

### PR DESCRIPTION
**Pull request by TonyValenti**
There were a number of places where the ```DataDirectory``` variable was being retrieved and then was having string replacements done.  I factored out these methods into their own classes and then updated the code so that everywhere calls the same methods.

### CODE REVIEW

File: Oqtane.Shared/Configuration/ConnectionString.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Oqtane.Configuration
+{
+    public static class ConnectionString
```
Comment (hishamco) [inline Oqtane.Shared/Configuration/ConnectionString.cs:9]: Static classes, noooooo

**Comment by hishamco** [comment]
Please @sbwalker don't merge this PR, there is a better way to make this happen without `static` classes

Thanks @TonyValenti for your contribution, but we should do it right way

**Comment by TonyValenti** [comment]
@hishamco What is your recommendation?

File: Oqtane.Server/Extensions/WebHostBuilderExtensions.cs [hunk 2]
```diff
@@ -12,7 +13,7 @@ public static IWebHostBuilder ConfigureLocalizationSettings(this IWebHostBuilder
             {
                 var config = context.Configuration;
 
-                services.Configure<LocalizationOptions>(config.GetSection("Localization"));
+                services.Configure<LocalizationOptions>(config.GetSection(SettingKeys.LocalizationSection));
```
Comment (hishamco) [inline Oqtane.Server/Extensions/WebHostBuilderExtensions.cs:16]: Seems this go beyond DataDirectory ;)

**Comment by hishamco** [comment]
> @hishamco What is your recommendation?

Perhaps using options pattern for all things related to `appsettings.json`

But my advice as always @sbwalker remind me, let us think about V2 features and backlog, coz such PRs needs a review and perhaps breaking things, I already did one in the past ;) so please let us focus on V2, if we finish we can refactor and enhance the code

Thanks

File: Oqtane.Server/Extensions/WebHostBuilderExtensions.cs (hunk 2 above)
Comment (TonyValenti) [reply Oqtane.Server/Extensions/WebHostBuilderExtensions.cs:16]: When I create a PR, it wraps in things in from subsequent commits as well.  You'll notice that is from the next commit.

**Comment by sbwalker** [comment]
I am not going to merge this PR as it touches DatabaseManager - which has at least 4 distinct scenarios related to installation/upgrade which would be affected and I am fairly certain that not all of these scenarios were tested with these changes.

**Comment by hishamco** [comment]
Agree with @sbwalker we may have a look after `2.0.0` with unit and functional tests provided to make sure nothing is broken

**Pull request closed without merging**